odoo/orm/models.py

```diff
@@ -3538,6 +3584,17 @@ def _setup_base(self):
         elif 'x_active' in cls._fields:
             cls._active_name = 'x_active'
 
+        # 7. determine table objects
+        table_objects = {}
+        for klass in reversed(cls._model_classes):
+            # this condition is an optimization of is_definition_class(klass)
+            if isinstance(klass, MetaModel):
+                for cons in klass._table_object_definitions:
+                    table_objects[cons.name] = cons
+        for cons in cls._table_object_definitions:
+            table_objects[cons.name] = cons
+        cls._table_objects = tuple(table_objects.values())
```

Comment:
As `cls` is a model's registry class, `cls._table_object_definitions` should be empty, and the whole loop could become a dict comprehension.